gpytorch/lazy/cat_lazy_tensor.py

```diff
@@ -0,0 +1,333 @@
+            if target_indices == slice(None, None, None):
+                res_list = [self._lazify(t._getitem(*indices)) for t in self.lazy_tensors]
+                return self.__class__(*res_list, dim=new_cat_dim, output_device=self.output_device)
+            else:
+                target_slices = self._split_slice(target_indices)
+                target_tensors = [self.idx_to_tensor_idx[sl.start] for sl in target_slices]
+
+                res_list = []
+                for idx, t_idx in zip(target_slices, target_tensors):
+                    shifted_start = idx.start - self.tensor_idx_to_start_idx[t_idx]
+                    shifted_stop = idx.stop - self.tensor_idx_to_start_idx[t_idx]
+                    shifted_slice = slice(shifted_start, shifted_stop, idx.step)
+                    indices[self.cat_dim] = shifted_slice
+                    res = self._lazify(self.lazy_tensors[t_idx]._getitem(*indices))
+                    res_list.append(res)
+                if len(res_list) == 1:
+                    return res_list[0]
+                elif all([rl.dim() == 1 for rl in res_list]):
+                    return torch.cat([rl.evaluate().to(self.device) for rl in res_list])
+                else:
+                    return self.__class__(*res_list, dim=new_cat_dim, output_device=self.output_device)
+        elif torch.is_tensor(target_indices):
+            # this means another `indices` is a slice object
+            target_indices = [idx.item() for idx in target_indices]
+            target_tensors = [self.idx_to_tensor_idx[idx] for idx in target_indices]
+
+            res_list = []
+            curr_tensor, slice_indices = target_tensors[0], []
+            for idx, t_idx in zip(target_indices, target_tensors):
+                if t_idx != curr_tensor:
+                    indices[self.cat_dim] = torch.tensor(slice_indices)
+                    res = self._lazify(self.lazy_tensors[t_idx]._getitem(*indices))
+                    res_list.append(res)
+                    curr_tensor, slice_indices = t_idx, []
+                slice_indices.append(idx - self.tensor_idx_to_start_idx[t_idx])
+            indices[self.cat_dim] = torch.tensor(slice_indices)
+            res = self._lazify(self.lazy_tensors[t_idx]._getitem(*indices))
+            res_list.append(res)
+            return self.__class__(*res_list, dim=new_cat_dim, output_device=self.output_device)
+
+    def _get_indices(self, left_indices, right_indices, *batch_indices):
+        # tensor indices must all have the same length
+        indices = list(batch_indices) + [left_indices, right_indices]
+        indices = torch.stack(indices, dim=0)
+        target_indices = indices[self.cat_dim, :]
+        target_tensors = [self.idx_to_tensor_idx[idx.item()] for idx in target_indices]
+        starting_indices = [self.tensor_idx_to_start_idx[t_idx] for t_idx in target_tensors]
+        local_indices = target_indices - torch.tensor(starting_indices)
```

Comment:
@jacobrgardner are you sure the current version runs on multiple GPUs? I'm getting an error here when I try to run the GP Regression with LOVE notebook. It seems like `target_indices` and `torch.tensor(starting_indices)` can sometimes live on different devices.